pkgs/development/libraries/libcamera/default.nix

```diff
@@ -0,0 +1,85 @@
+{ stdenv
+, fetchgit
+, lib
+, meson
+, ninja
+, pkg-config
+, boost
+, gnutls
+, openssl
+, libevent
+, lttng-ust
+, libtiff
+, qt5
+, gst_all_1
+, graphviz
+, doxygen
+, python3
+, python38Packages
+}:
+
+stdenv.mkDerivation {
+  pname = "libcamera";
+  version = "1.0.0";
+
+  src = fetchgit {
+    url = "git://linuxtv.org/libcamera.git";
+    rev = "040a6dcfbccaf1005da2c571769b1f89df1f07cf";
+    sha256 = "sha256-RCNKFCv3fNabroCUR5MtKbyDS20J7leU8KkE1U5qe1M=";
+  };
+
+  patchPhase = ''
+    patchShebangs utils/
+  '';
+
+  buildInputs = [
+    # core requirements
+    python38Packages.pyyaml
+    python38Packages.ply
+    python38Packages.jinja2
+
+    # IPA and signing
+    gnutls
+    openssl
+    boost
+
+    # gstreamer integration
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
```

Comment:
How does this integration work?

Is that idea that `libcamera` could be added to https://github.com/NixOS/nixpkgs/blob/3b6c3bee9174dfe56fd0e586449457467abe7116/pkgs/desktops/gnome/apps/cheese/default.nix#L81-L84 and Cheese would use libcamera as a gstreamer source?

Reply:
I believe so, yeah. Check out https://libcamera.org/getting-started.html#using-gstreamer-plugin

Reply:
Confirmed that gstreamer is able to find and use this.